Update queries.js

jpgpi250 opened the pull request:
Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

accompanies FTL PR 1409: Store domainlist IDs for blocked/permitted queries

## Thank you for your contribution to the Pi-hole Community! 

Please read the comments below to help us consider your Pull Request.

We are all volunteers and completing the process outlined will help us review your commits quicker.

**Please make sure you**
 1. Base your code and PRs against the repositories developmental branch.
 2. [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits as we enforce the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions

---
- **What does this PR aim to accomplish?:**

accompanies FTL PR 1409, create links for domainlist IDs in query log view

- **How does this PR accomplish the above?:**

modify queries.js in AdminLTE\scripts\pi-hole\js. Check additional_data field for all queries, originally only status 4 & 10

- **What documentation changes (if any) are needed to support this PR?:**

already documented by DL6ER: https://github.com/pi-hole/docs/pull/774

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
- [x] I have read the above and my PR is ready for review. _Check this box to confirm_

jpgpi250 committed:
Update queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

accompanies FTL PR 1409

jpgpi250 committed:
Update queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

accompanies FTL PR 1409

jpgpi250 committed:
Update queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

jpgpi250 committed:
Update queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

DL6ER commented on scripts/pi-hole/js/queries.js:
```diff
@@ -115,6 +115,11 @@ $(function () {
         regexLink = false;
 
+      // accompanies Store domainlist IDs for blocked/permitted queries FTL PR 1409
+      if (data.length > 9 && data[9] > 0) {
```
We should check if the `data[9]` field is numeric - could also be a CNAME domain or empty- before checking if it is greater than zero:

```suggestion
      if (data.length > 9 && is_numeric(data[9]) && data[9] > 0) {
```

Note: I am on a business trip and haven't been able to actually test this suggestion.

jpgpi250 committed:
Update scripts/pi-hole/js/queries.js

Co-authored-by: DL6ER <DL6ER@users.noreply.github.com>
Signed-off-by: Peter Russell <jpgpi250@gmail.com>

jpgpi250 committed:
Update scripts/pi-hole/js/queries.js

Co-authored-by: DL6ER <DL6ER@users.noreply.github.com>
Signed-off-by: Peter Russell <jpgpi250@gmail.com>

jpgpi250 committed:
Update queries.js

'is_numeric', as suggested by DL6ER, is not defined.

replaced with '!isNaN'

jpgpi250 committed:
Update queries.js

'is_numeric', as suggested by DL6ER, is not defined.

replaced with '!isNaN'

jpgpi250 committed:
added comment & DCO

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

jpgpi250 committed:
added comment & DCO

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

jpgpi250 committed:
left padding

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

jpgpi250 committed:
left padding

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

jpgpi250 commented:
can't get DCO requirement anymore, forgot to signof in one of the commits.

@DL6ER: your suggestion 'is_numeric' doesn't work (undefined)
duckduckgo suggests several methods to implement numeric test, [here](https://orangeable.com/javascript/is-numeric) and [here](https://flaviocopes.com/how-to-check-value-is-number-javascript/)

tested these methods, '!isNAN' appears to be the only one working.

Note that there weren't any problems with the original code (no numeric test) an everything worked as expected:
- additional info for CNAME entries displayed correctly.
- link provided, if the additional_info field contained a domainlistID

I'm have virtually no javascript skills, up to you people to decide which test is required, if at all.

yubiuser commented:
DCO is not a problem. We can set it to pass manually.

jpgpi250 commented:
tested with w3schools to ensure !isNAN works

![image](https://user-images.githubusercontent.com/24669386/189305761-158a5722-2786-40ab-9c84-a06276c7106e.png)

yubiuser commented:
`is_numeric` is a php function, not a js.
https://www.php.net/manual/de/function.is-numeric.php

yubiuser commented on scripts/pi-hole/js/queries.js:
```diff
@@ -115,6 +115,12 @@ $(function () {
         regexLink = false;
 
+      // accompanies Store domainlist IDs for blocked/permitted queries FTL PR 1409
+      // 'is_numeric' is undefined, replaced with '!isNAN', see PR for details
```
```suggestion

```

yubiuser commented on scripts/pi-hole/js/queries.js:
```diff
@@ -115,6 +115,12 @@ $(function () {
 
+      // accompanies Store domainlist IDs for blocked/permitted queries FTL PR 1409
+      // 'is_numeric' is undefined, replaced with '!isNAN', see PR for details
+      if (data.length > 9 && !isNaN(data[9]) && data[9] > 0) {
```
I don't like double negatives.
```suggestion
      if (data.length > 9 && Number.isInteger(data[9]) && data[9] > 0) {
```

jpgpi250 replied:
I'm sorry, I did test this, since it was suggested in one of the duckduckgo results, linked (above), however, it doesn't work, tested with firefox and edge chromium.

result (no link) with if (data.length > 9 && Number.isInteger(data[9]) && data[9] > 0) {:

![image](https://user-images.githubusercontent.com/24669386/189331447-f7b285ac-d36f-4670-8dc4-b4a5a00871bc.png)

result (link) with if (data.length > 9 && !isNaN(data[9]) && data[9] > 0) {

![image](https://user-images.githubusercontent.com/24669386/189331672-c5bb89e9-3ede-470b-b0b2-0ae4d18a9d78.png)

yubiuser replied:
Try `Number.isInteger(parseInt(data[9],10))`

jpgpi250 replied:
That works, but again, aren't we over complicating things? Remember that everything worked fine without the numeric test. we now add 'Number.isInteger' and 'parseInt(data[9]) for every query (all status types)...

jpgpi250 committed:
Update scripts/pi-hole/js/queries.js

Co-authored-by: yubiuser <ckoenig@posteo.de>
Signed-off-by: Peter Russell <jpgpi250@gmail.com>

jpgpi250 committed:
Update scripts/pi-hole/js/queries.js

Co-authored-by: yubiuser <ckoenig@posteo.de>
Signed-off-by: Peter Russell <jpgpi250@gmail.com>

jpgpi250 committed:
Update scripts/pi-hole/js/queries.js

Co-authored-by: yubiuser <ckoenig@posteo.de>
Signed-off-by: Peter Russell <jpgpi250@gmail.com>

jpgpi250 committed:
Update scripts/pi-hole/js/queries.js

Co-authored-by: yubiuser <ckoenig@posteo.de>
Signed-off-by: Peter Russell <jpgpi250@gmail.com>

jpgpi250 committed:
Update queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

latest suggestion from yubiuser

jpgpi250 committed:
Update queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

latest suggestion from yubiuser

jpgpi250 committed:
Update queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

test failed, additional space required

jpgpi250 committed:
Update queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

test failed, additional space required

rdwebdesign commented:
You forgot DCO here: [d31aec1](https://github.com/pi-hole/AdminLTE/pull/2341/commits/d31aec156ba0bb328e3dea73aa37ce78b5cc6d4e)

You can fix this using `git rebase -i`. Choose option `reword` and add your `Signed-off-by:` info.

If you need help, these links explain the process:
- https://matthewsetter.com/git-interactive-rebase/
- https://medium.com/dana-engineering/rewriting-history-with-git-rebase-interactive-adf0341afa93#e0d5

jpgpi250 commented:
@rdwebdesign 

Using github desktop here, on windows. I use very few functions of github desktop, modifying files using my favorite editor. I haven't found these options in github desktop, maybe I'm overlooking something.

rdwebdesign commented:
Hmmmm... I never used github desktop.
I use the command line to run `git rebase` and the file opens in `vscode`.

Note: we will probably set DCO to pass.

jpgpi250 commented:
@rdwebdesign

there is an entry in the menu of github desktop, but I have no idea what this does (and I'm always afraid to f*** things up).
I enter my signoff message (Signed-off-by: jpg...) in the description field (bottom / left) and commit (button). this usually does the trick, unless I forget the signoff -> DCO error.

![image](https://user-images.githubusercontent.com/24669386/189428161-1c77467f-8c3c-4835-9043-86215fa1c68a.png)

rdwebdesign commented:
No!
This is a different `rebase` operation.

Maybe there is no option to do it using Github Desktop (as I said, I don't use it).

jpgpi250 commented:
@rdwebdesign 
in github (browser, where I'm typing this) I see this:

![image](https://user-images.githubusercontent.com/24669386/189429614-b7ad8495-dc19-4e4e-a78c-89aa9c028346.png)

any idea what this means, and does it require an action from me?

Sorry, I try to contribute, but it's all very new to me, and the suggested changes haven't made it any easier for me to keep up.

Thanks for time, help and effort.

rdwebdesign commented:
This is the same operation you were seeing in Github Desktop menu (I guess). This will update your branch.

`devel` branch has new code (created after you started your branch) and you need to update your branch, so they will be "in sync".

There are 2 ways to do it:
- _with Merge_: it will merge the new code on top of your changes;
- _with Rebase_: it will move your branch to the tip of the `devel` code. Your code will be on top of the current `devel.`

You can do it using Github Desktop or github web interface.

I suggest you to use "with Rebase".

jpgpi250 commented:
@rdwebdesign 
```
I suggest you to use "with Rebase".
```
- done that, but the Code scanning results / CodeQL check took 75 minutes.
- checked 'files changed', the code matches what I'm currently testing (on my pi)
- the code currently in the PR is the latest suggestion from yubiuser, e.g.
```
if (data.length > 9 && Number.isInteger(parseInt(data[9], 10)) && data[9] > 0) {
```
- don't know if this will ever go into dev, all depends on what happens in [FTL PR 1409](https://github.com/pi-hole/FTL/pull/1409)

jpgpi250 committed:
Merge branch 'devel' of https://github.com/jpgpi250/AdminLTE into devel

rdwebdesign commented:
I'm not sure I understood your second and third items above.

If you used the github web interface to update your branch, 2 things happened:
- github rebased your branch;
- github force pushed the commits, rewriting the commits order.

In this case, your local code will be different than github and you will need to pull the changes.

yubiuser commented on scripts/pi-hole/js/queries.js:
```diff
@@ -115,6 +115,11 @@ $(function () {
 
+      // accompanies Store domainlist IDs for blocked/permitted queries FTL PR 1409
+      if (data.length > 9 && Number.isInteger(parseInt(data[9], 10)) && data[9] > 0) {
+        regexLink = true;
```
```suggestion
        DomainlistLink = true;
```

rdwebdesign replied:
Suggestions:
`listLink` or `itemLink`

rdwebdesign replied:
The suggestion here, was to replace ALL instances of `regexLink` with a different name.

You can choose any name. 
We suggested to use `DomainlistLink`, or `listLink`, or `itemLink` (or whatever you like).

jpgpi250 replied:
```
The suggestion here, was to replace ALL instances of regexLink with a different name.
```
I already did that, and I can see the changes in "Files Changed", 'DomainlistLink' is there 3 times. I've also replaced the modified file on my pihole, everything appears to be working.

![image](https://user-images.githubusercontent.com/24669386/189581548-9639dba4-1a6a-43e5-8a13-e6b727b4124c.png)

![image](https://user-images.githubusercontent.com/24669386/189581687-6c037a7c-dd1b-43a4-8e68-c3730fa7acf3.png)

yubiuser commented on scripts/pi-hole/js/queries.js:
```diff
@@ -115,6 +115,11 @@ $(function () {
         isCNAME = false,
         regexLink = false;
```
```suggestion
        DomainlistLink = false;
```

yubiuser reviewed (commented):
I would rename `regexLink` to something else, as it is not limited to regex anymore.
This would also need change here (including change to the tooltip):
https://github.com/pi-hole/AdminLTE/blob/37dbfc8ee667f3c09511ab696bcb3a7cc61c4ac4/scripts/pi-hole/js/queries.js#L245

yubiuser commented:
`DomainlistLink` is just a suggestion maybe you got a nicer phrase.

jpgpi250 committed:
Update scripts/pi-hole/js/queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

Co-authored-by: yubiuser <ckoenig@posteo.de>
Signed-off-by: Peter Russell <jpgpi250@gmail.com>

jpgpi250 committed:
Update scripts/pi-hole/js/queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

Co-authored-by: yubiuser <ckoenig@posteo.de>
Signed-off-by: Peter Russell <jpgpi250@gmail.com>

jpgpi250 committed:
Update queries.js

Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

I assume this needded to be changed to (line 242 regexLink -> DomainlistLink). running this version now & ftl development

jpgpi250 committed:
Merge branch 'devel' into devel

yubiuser commented on scripts/pi-hole/js/queries.js:
```diff
@@ -242,7 +239,7 @@ $(function () {
+      if (DomainlistLink) {
         $("td:eq(4)", row).hover(
           function () {
             this.title = "Click to show matching regex filter";
```
```suggestion
            this.title = "Click to show matching blacklist/whitelist domain";
```

jpgpi250 committed:
Signed-off-by: jpgpi250 <jpgpi250@gmail.com>

Co-authored-by: yubiuser <ckoenig@posteo.de>
Signed-off-by: Peter Russell <jpgpi250@gmail.com>

yubiuser commented:
Looks good to me now. 
Now the hardest part follows: you need to squash your 26 commits locally and force push the remaining commit.

jpgpi250 commented:
```
Now the hardest part follows: you need to squash your 26 commits locally and force push the remaining commit.
```
neither github desktop, nor the browser indicate there is something left to do.

doesn't appear to be possible (button is not enabled)
![image](https://user-images.githubusercontent.com/24669386/189592105-2d83ebcc-25b3-4c31-be6f-709ba48b55c0.png)

yubiuser commented:
Do it in the terminal:
`git rebase -i  HEAD~26`

https://www.git-tower.com/learn/git/faq/git-squash

jpgpi250 commented:
I'm using github desktop on a windows computer, git not installed.
to make things easier, I've added you as contributor to the repository (https://github.com/jpgpi250/AdminLTE)

yubiuser commented:
Have a look here: https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/managing-commits/squashing-commits

yubiuser commented:
> If a group of commits represents a single unit of work, or if a project requires that each pull request only has one commit, simply drag them on top of one another to squash them together and add a new commit message that captures the whole picture.

https://github.blog/2021-06-16-github-desktop-2-9-squashing-reordering-amending/#squash-and-reorder-commits

jpgpi250 commented:
cloned onto my pi, then 'git rebase -i HEAD~26'
```
[detached HEAD c86962d1] Only add link to Query Log when filtering property is available
 Author: DL6ER <dl6er@dl6er.de>
 Date: Tue Jan 19 07:11:11 2021 +0100
 15 files changed, 6474 insertions(+), 54 deletions(-)
 create mode 100644 style/themes/default-darker.css
Auto-merging scripts/pi-hole/php/savesettings.php
CONFLICT (content): Merge conflict in scripts/pi-hole/php/savesettings.php
Auto-merging scripts/pi-hole/php/groups.php
error: could not apply 03920e35... Move validation functions into func.php and use the same validation in all cases.
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 03920e35... Move validation functions into func.php and use the same validation in all cases.
```

yubiuser commented:
As I said: it's the hardest part...

You ran into merge conflicts. You could try to solve them, or probably easier (because only a few lines changed) start fresh. I'm fine with the current state of your PR and approve right away.
